test/unit/storage/FileResourceStore.test.ts

```diff
@@ -0,0 +1,225 @@
+    await expect(store.getRepresentation({ path: 'http://wrong.com/wrong' })).rejects.toThrow(NotFoundHttpError);
+    await expect(store.getRepresentation({ path: `${base}wrong` })).rejects.toThrow(NotFoundHttpError);
+    await expect(store.addResource({ path: 'http://wrong.com/wrong' }, representation))
+      .rejects.toThrow(NotFoundHttpError);
+    await expect(store.deleteResource({ path: 'wrong' })).rejects.toThrow(NotFoundHttpError);
+    await expect(store.setRepresentation({ path: 'http://wrong.com/' }, representation))
+      .rejects.toThrow(NotFoundHttpError);
+  });
+
+  it('errors when modifying resources.', async(): Promise<void> => {
+    await expect(store.modifyResource()).rejects.toThrow(Error);
+  });
+
+  it('errors for wrong input data types.', async(): Promise<void> => {
+    (representation as any).dataType = DATA_TYPE_QUAD;
+    await expect(store.addResource({ path: base }, representation)).rejects.toThrow(UnsupportedMediaTypeHttpError);
+  });
+
+  it('can write and read a container.', async(): Promise<void> => {
+    // Mock the fs functions.
+    fs.promises.mkdir = jest.fn();
+    (fs.promises.mkdir as jest.Mock).mockReturnValue(true);
+    fs.promises.lstat = jest.fn();
+    stats.isDirectory = (): any => true;
+    (fs.promises.lstat as jest.Mock).mockReturnValueOnce(stats);
+    stats.isFile = (): any => true;
+    (fs.promises.lstat as jest.Mock).mockReturnValueOnce(stats);
+    const quad = triple(
+      namedNode(`${base}myContainer/`),
+      namedNode('http://www.w3.org/1999/02/22-rdf-syntax-ns#type'),
+      namedNode('http://www.w3.org/ns/ldp#BasicContainer'),
+    );
+    fs.createReadStream = jest.fn();
+    (fs.createReadStream as jest.Mock).mockReturnValueOnce(streamifyArray([ quad ]));
+    (fs.createReadStream as jest.Mock).mockImplementation((): any => new Error('Metadata file does not exist.'));
+
+    // Write container (POST)
+    (representation as any).metadata = { linkType: LINK_TYPE_LDP_BC, slug: 'myContainer/' };
+    const identifier = await store.addResource({ path: base }, representation);
+    expect(identifier.path).toBe(`${base}myContainer/`);
+
+    // Read container
+    const result = await store.getRepresentation(identifier);
+    expect(result).toEqual({
+      dataType: DATA_TYPE_BINARY,
+      data: expect.any(Readable),
+      metadata: {
+        profiles: [],
+        raw: [],
+        dateTime: stats.mtime,
+        byteSize: stats.size,
+      },
+    });
+    await expect(arrayifyStream(result.data)).resolves.toEqualRdfQuadArray([ quad ]);
+  });
+
+  it('errors for container creation with path to non container.', async(): Promise<void> => {
+    // Mock the fs functions.
+    fs.promises.mkdir = jest.fn();
+    (fs.promises.mkdir as jest.Mock).mockReturnValue(true);
+    fs.promises.lstat = jest.fn();
+    stats.isDirectory = (): any => false;
+    (fs.promises.lstat as jest.Mock).mockReturnValue(stats);
```

Comment:
Probably don't want to have to repeat that.

If you do, it likely belongs in a `before`  so that the `after` can clean up (otherwise, failing tests will lead to weird effects in other tests).